.github/workflows/build.yaml

```
@@ -9,6 +9,77 @@ on:
     branches: [ master ]
 
 jobs:
+  changes:
+    name: changed files
+    runs-on: ubuntu-latest
+    # Set job outputs to values from changed_files step for other jobs
+    outputs:
+      codecheck: ${{ steps.changed_files.outputs.codecheck }}
+      clang_tidy: ${{ steps.changed_files.outputs.clang_tidy }}
+      run_test_suite: ${{ steps.changed_files.outputs.test_suite }}
+      inst_windows_msvc: ${{ steps.changed_files.outputs.inst_windows_msvc }}
+      inst_windows: ${{ steps.changed_files.outputs.inst_windows }}
+      inst_macos: ${{ steps.changed_files.outputs.inst_macos }}
+      appimage: ${{ steps.changed_files.outputs.appimage }}
+    steps:
+    - name: Check for changed files
+      uses: dorny/paths-filter@v2
+      id: changed_files
+      with:
+        # groups to check for changed files (as string, yaml anchors are resolved by the tool)
+        filters: |
+          codecheck:
+            - 'CMakeLists.txt'
+            - '*/CMakeLists.txt'
+            - 'cmake/codecheck/**'
+            - 'cmake/WlFunctions.cmake'
+            - 'src/**'
+            - '.github/scripts/install_deps.sh'
```

```suggestion
            - *compile_and_dependencies
```
except it'sdefined later